### PR TITLE
Serverless function improvements

### PR DESCRIPTION
**Pull request by martmull**
- add layer for lambda execution
- add layer for local execution
- add package resolve for the monaco editor
- add route to get installed package for serverless functions
- add layer versioning

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/input/code-editor/components/CodeEditor.tsx [hunk 1]
```diff
@@ -38,12 +41,23 @@ export const CodeEditor = ({
...
+  const { availablePackages } = useGetAvailablePackages();
+
+  const handleEditorDidMount = async (
     editor: editor.IStandaloneCodeEditor,
     monaco: Monaco,
   ) => {
     monaco.editor.defineTheme('codeEditorTheme', codeEditorTheme(theme));
     monaco.editor.setTheme('codeEditorTheme');
+
+    if(language === 'typescript') {
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/ui/input/code-editor/components/CodeEditor.tsx:53]:
```suggestion
    if (language === 'typescript') {
```

File: packages/twenty-front/src/modules/ui/input/code-editor/components/CodeEditor.tsx [hunk 2]
```diff
@@ -68,28 +82,31 @@ export const CodeEditor = ({
...
-            horizontal: 'hidden',
-          },
-          minimap: {
-            enabled: false,
-          },
-        }}
-      />
-    </div>
+    isDefined(availablePackages) && (
+      <div>
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/ui/input/code-editor/components/CodeEditor.tsx:88]:
```suggestion
      <>
```

File: packages/twenty-front/src/pages/settings/serverless-functions/SettingsServerlessFunctionDetail.tsx [hunk 3]
```diff
@@ -72,7 +68,7 @@ export const SettingsServerlessFunctionDetail = () => {
     }
   };
 
-  const handleSave = useDebouncedCallback(save, 500);
+  const handleSave = usePreventOverlapCallback(save, 1000);
```
Comment (thomtrp) [inline packages/twenty-front/src/pages/settings/serverless-functions/SettingsServerlessFunctionDetail.tsx:71]: so the goal of this function was preventing the double save?

File: packages/twenty-server/src/engine/integrations/serverless/drivers/local.driver.ts [hunk 4]
```diff
@@ -39,22 +41,40 @@ export class LocalDriver
     this.fileStorageService = options.fileStorageService;
   }
 
+  private getInMemoryLayerFolderPath = (version: number) => {
+    return join(SERVERLESS_TMPDIR_FOLDER, COMMON_LAYER_NAME, `${version}`);
+  };
+
+  private async createLayerIfNotExists(version: number) {
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/integrations/serverless/drivers/local.driver.ts:48]: @martmull I would like to discuss this with you. A layer is not really specific enough to me. When we say a lambda layer, we know we are talking about modules management. But I am not sure this naming should be used for the local driver. 

I feel we need something more generic there, related to modules management.

File: packages/twenty-front/src/pages/settings/serverless-functions/SettingsServerlessFunctionDetail.tsx (hunk 3 above)
Comment (martmull) [reply packages/twenty-front/src/pages/settings/serverless-functions/SettingsServerlessFunctionDetail.tsx:71]: yes, save can take several seconds when using lambdas, and it returns an error if you update an updating function

File: packages/twenty-server/src/engine/integrations/serverless/drivers/local.driver.ts (hunk 4 above)
Comment (martmull) [reply packages/twenty-server/src/engine/integrations/serverless/drivers/local.driver.ts:48]: I used layer to match the aws naming.
For example, when we will use python serverless functions, the "module" naming won't fit as it is called "requirements" in python

**Review by thomtrp — APPROVED** [review]

**Pull request merged**